File: archive/p/perl/fibonacci.pl
#!/usr/bin/env perl
use v5.42;

sub usage {
    say "Usage: please input the count of fibonacci numbers to output";
    exit;
}

my ($n) = @ARGV;
usage() unless defined $n && $n =~ /\A\d+\z/;

my ( $a, $b ) = ( 0, 1 );

for my $i ( 1 .. $n ) {
    ( $a, $b ) = ( $b, $a + $b );
    say "$i: $a";
}

File: archive/p/pascal/fibonacci.pas
program Fibonacci(input, output, stdErr);
(*Read count of fibonnaci numbers into a string

*)
var
  buf: String;
  count, check, first,second,result, i : integer;

begin
  (*Variable initialisation must be inside begin-end block*)
  first := 0;
  second := 1;
  result := 0;

  begin   
   buf:= paramStr(1);
   Val(buf, count, check);  
   (*If input is valid integer, check will be 0, else will be 1*)
   if (check <> 0)
   then
   begin
    writeln('Usage: please input the count of fibonacci numbers to output');
   end
   else
    for i := 1 to count do
    begin
    
    result := first + second;
    first := second;
    second := result;
    writeln(i, ': ', first);
    end;
   end;
end.
